nix/postgres.nix

```diff
@@ -136,6 +136,22 @@ in
           '';
         };
 
+        defaultSettings =
+          lib.mkOption {
+            type = lib.types.raw;
+            internal = true;
+            readonly = true;
+            description = ''
+              Default configuration for `postgresql.conf`. `settings` can override these values.
+            '';
+            default = {
+              listen_addresses = config.listen_addresses;
+              port = config.port;
+              unix_socket_directories = lib.mkDefault config.dataDir;
```

Comment:
Is `lib.mkDefault` necessary here?

Reply:
removed it! We don't care much about the priority of the default here anyway.